sklearn/utils/_testing.py

```diff
@@ -1151,3 +1151,22 @@ def transform(self, X, y=None):
 
     def fit_transform(self, X, y=None):
         return self.fit(X, y).transform(X, y)
+
+
+def assert_radius_neighborhood_results_equality(
+    ref_dist, dist, ref_indices, indices, radius
+):
```

Comment:
Since it's not a generic testing tool but specific to neighbors, I think it was better located in test_pairwise_distances_reduction.py